07.md

```diff
@@ -18,8 +18,10 @@ async window.nostr.signEvent(event: Event): Event // takes an event object, adds
 Aside from these two basic above, the following functions can also be implemented optionally:
 ```
 async window.nostr.getRelays(): { [url: string]: {read: boolean, write: boolean} } // returns a basic map of relay urls to relay policies
-async window.nostr.nip04.encrypt(pubkey, plaintext): string // returns ciphertext and iv as specified in nip-04
-async window.nostr.nip04.decrypt(pubkey, ciphertext): string // takes ciphertext and iv as specified in nip-04
+async window.nostr.nip04.encrypt(pubkey, plaintext): string // returns ciphertext and iv as specified in nip-04 (deprecated)
+async window.nostr.nip04.decrypt(pubkey, ciphertext): string // takes ciphertext and iv as specified in nip-04 (deprecated)
+async window.nostr.nip44.encrypt(sender_privkey, recipient_pubkey, plaintext, version): string // returns encrypted payload as specified in nip-44
```

Comment:
private key is added with third party pubkey and then used to calculate shared key. There is no other way around this. My usual suggestion with regards to `window.nostr` is to expose `getConversationKey` or so, which is like your getKey

Reply:
yes but nostr window object should already have the private key, so we shouldn't need to submit it as part of the function call. right? the client using this function call won't even know the private key, thats why it is calling the nostr object to encrypt. or is there a different private key that this is referring to?

Reply:
07.md must be updated. we have just 3 methods: `get_conv_key(privA, pubB)`, `encrypt(text, conv_key)`, `decrypt(text, conv_key)`

Reply:
Removed stuff from NIP-07 until you reach an agreement.
ff533d7a99715b0524f93a182f6c2169fe3b25b1

It definitely can't be a function that asks for your private key otherwise it's better to not have it at all.

Reply:
whatever we choose, it should be consistent with nip46 methods (in naming, args, and functionality).

my preference would be to use the methods as they are structured in nip46.

> 07.md must be updated. we have just 3 methods: `get_conv_key(privA, pubB)`, `encrypt(text, conv_key)`, `decrypt(text, conv_key)`

this doesn't make sense to me. any app that calls get `get_conv_key` wouldn't need to call `encrypt` or `decrypt`, because it already has the conversation key to do the encryption/decryption. encrypt/decrypt should take the 3rd party pubkey because:
- if they take the conversation key these functions are completely unnecessary
- it gives apps the option not to store the conversation key